ui/components/ui/delineator/utils.ts

```diff
@@ -0,0 +1,66 @@
+import React from 'react';
+import {
+  BackgroundColor,
+  IconColor,
+  TextColor,
+} from '../../../helpers/constants/design-system';
+import { AvatarIconSize, IconProps, Text } from '../../component-library';
+import { DelineatorType } from './delineator.types';
+
+const defaultIconProps = {
+  size: AvatarIconSize.Xs,
+};
+
+/*
+ * Get the icon props based on the type of delineator.
+ */
+export const getIconPropsByType = (type?: DelineatorType) => {
+  let iconProps = {};
+  const inverseIconColorProp = {
+    color: IconColor.infoInverse,
+  } as IconProps<'span'>;
+
+  switch (type) {
+    case DelineatorType.Error:
+      iconProps = {
+        iconProps: inverseIconColorProp,
+        backgroundColor: BackgroundColor.errorDefault,
+      };
+      break;
+    default:
+      iconProps = {
+        iconProps: inverseIconColorProp,
+        backgroundColor: BackgroundColor.overlayAlternative,
+      };
+  }
+  return { ...defaultIconProps, ...iconProps };
+};
+
+/*
+ * Get the text color based on the type of delineator.
+ */
+const getTextColorByType = (type?: DelineatorType) => {
+  switch (type) {
+    case DelineatorType.Error:
+      return TextColor.errorDefault;
+    default:
+      return TextColor.textAlternative;
+  }
+};
```

Comment:
Nit: Above function can be a simple object key -> value could be DelineatorType -> TextColor

Reply:
Good call but then default value needs to be determined by an `||` operator. 
```
const textColorMap = {
  [DelineatorType.Error]: TextColor.errorDefault,
};

/// usage
textColorMap[type] || TextColor.textAlternative;
```